release/nightly_tests/stress_tests/test_state_api_scale.py

```diff
@@ -121,7 +121,7 @@ def test_many_actors(num_actors: int):
         logger.info("Skipping test with no actors")
         return
 
-    @ray.remote
+    @ray.remote(concurrency_groups={"io": 1})
```

Comment:
This is the test that catches the issue that caused the CI failure in https://buildkite.com/ray-project/premerge/builds/36904#0195b532-f96f-42cc-8bfb-4fb904054433

Reply:
In addition, the reproduction script in the issue description is almost the same as this test.

Reply:
Is this test correct? I think normal actor should also be tested right? You changed the normal actor to threaded actor.

Reply:
I am also wondering if only 10 actors can catch this error

https://github.com/ray-project/ray/blob/b0828fb39e151ac836c007178feb494712f3f205/release/nightly_tests/stress_tests/test_state_api_scale.py#L376

Reply:
> Is this test correct? I think normal actor should also be tested right? You changed the normal actor to threaded actor.

https://github.com/ray-project/ray/pull/51582#issuecomment-2742505458

As stated in this comment, this is a blocker for #51582. Eventually, we will no longer have "normal" actors.

> I am also wondering if only 10 actors can catch this error. We may consider to increase the number of actors.

It fails three times (100% failure) consecutively on CI. https://buildkite.com/ray-project/premerge/builds/36904#0195b532-f96f-42cc-8bfb-4fb904054433. I can increase the number if you're concerned.

Reply:
I updated the tests. PTAL.

Reply:
I would still suggest paramterizing the test to run both with and without a concurrency group

Reply:
> I would still suggest paramterizing the test to run both with and without a concurrency group

@edoakes This seems to be how it is now. Would you mind explaining how it differs from the current PR?

Reply:
Hm what am I missing? The test doesn't have a `parametrize` decorator and there is only one `TestActor` definition.

I'm suggesting you do:
```
@pytest.mark.paramterize("use_concurrency_group", [False, True])
def test_many_actors(use_concurrency_group: bool):
    ...
    @ray.remote(concurrency_groups={"io": 1} if use_concurrency_group else None)
```

Reply:
I am a bit confused. The screenshot below shows the current test.

<img width="640" alt="image" src="https://github.com/user-attachments/assets/1365b29c-71f5-466c-a9a9-2be9390a4ead" />

Reply:
so weird... when I click on "view this file" from this thread it must be showing me an outdated state

Reply:
anyways, looks good :)